src/main/java/seedu/address/model/util/DateUtil.java

```diff
@@ -132,44 +144,63 @@ public static boolean compareDates(Date date1, Date date2) {
         }
     }
 
+    /**
+     * generate a date that is 1 month behind current date
+     */
+    public static Date generateOutdatedDebtDate(Date currentDate) {
+        String currentDateString = formatDate(currentDate);
+        int year = Integer.parseInt(currentDateString.substring(18, 22));
+        int day = Integer.parseInt(currentDateString.substring(5, 7));
+        int month = getMonthFromString(currentDateString.substring(8, 11));
+        if (month == JAN) { // if it is the month of Jan.
+            month = DEC; // set the month to Dec
+            year -= 1;
+        } else {
+            month -= 1; // go to prev month
+        }
+        month -= 1; // month is zero based
+        Date dateToReturn = new GregorianCalendar(year, month, day).getTime();
+        return dateToReturn;
+    }
```

Comment:
To get outdated date:

Calendar cal = Calendar.getInstance();
cal.setTime(currentDate)
cal.add(Calendar.MONTH, -1);
Date result = cal.getTime();

This wont work?

Reply:
Oh i think that would work too.